contrib/generic_proxy/filters/network/source/config.cc

```diff
@@ -19,11 +88,23 @@ Factory::createFilterFactoryFromProtoTyped(const ProxyConfig& proto_config,
           SINGLETON_MANAGER_REGISTERED_NAME(generic_route_config_provider_manager),
           [&context] { return std::make_shared<RouteConfigProviderManagerImpl>(context.admin()); });
 
-  const auto config =
-      std::make_shared<FilterConfig>(proto_config, context, *route_config_provider_manager);
+  auto factories = factoriesFromProto(proto_config.codec_config(), context);
+  std::shared_ptr<FilterFactory> custom_filter_factory = std::move(factories.second);
+
+  const FilterConfigSharedPtr config = std::make_shared<FilterConfigImpl>(
+      proto_config.stat_prefix(), std::move(factories.first),
+      routeConfigProviderFromProto(proto_config, context, *route_config_provider_manager),
+      filtersFactoryFromProto(proto_config.filters(), proto_config.stat_prefix(), context));
+
   return [route_config_provider_manager, config,
-          &context](Envoy::Network::FilterManager& filter_manager) -> void {
-    filter_manager.addReadFilter(std::make_shared<Filter>(config, context));
+          custom_filter_factory](Envoy::Network::FilterManager& filter_manager) -> void {
+    // Create filter by the custom filter factory if the custom filter factory is not null.
+    if (custom_filter_factory != nullptr) {
+      custom_filter_factory->createFilter(filter_manager, config);
+      return;
+    }
+
+    filter_manager.addReadFilter(std::make_shared<Filter>(config));
```

Comment:
Tips for reviewers.
If an extensible custom filter factory is existed, try it create read filter based on the factory.

Reply:
A question for the use case, I'm a little confused about this. If the developer creates a custom filter here, then it seems he will re-implement the major logic of the generic proxy filter? Or we expect the developer to create a custom filter that inherits from the generic proxy and do some customization?

Reply:
both are possible. But the second is what I think the major scene.